### ISSUE TITLE
[Bug]: Milvus 2.3.0 cannot use externalKafka with externalKafka.securityProtocol:SASL_SSL

### ISSUE THREAD

**Issue by weiZhenkun** [milvus-io/milvus#26599]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: 2.3.0
- Deployment mode(standalone or cluster): cluster
- MQ type(rocksmq, pulsar or kafka):   kafka 
- SDK version(e.g. pymilvus v2.0.0rc2): N/A
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

startup failed.

### Expected Behavior

startup successful.

### Steps To Reproduce

```markdown
externalKafka:
  enabled: true
  brokerList: "*****.amazonaws.com:9096,*****.amazonaws.com:9096,*****.amazonaws.com:9096"
  securityProtocol: SASL_SSL
  sasl:
    mechanisms: SCRAM-SHA-512
    username: "username"
    password: "password"
```

### Milvus Log

[2023/08/24 06:02:49.613 +00:00] [ERROR] [kafka/kafka_client.go:103] ["create sync kafka producer failed"] [error="Unsupported value \"SASL_SSL\" for configuration property \"security.protocol\": OpenSSL not available at build time"] [stack="github.com/milvus-io/milvus/pkg/mq/msgstream/mqwrapper/kafka.(*kafkaClient).getKafkaProducer.func1\n\t/go/src/github.com/milvus-io/milvus/pkg/mq/msgstream/mqwrapper/kafka/kafka_client.go:103\ngithub.com/milvus-io/milvus/pkg/util/conc.(*Singleflight[...]).Do.func1\n\t/go/src/github.com/milvus-io/milvus/pkg/util/conc/singleflight.go:19\ngolang.org/x/sync/singleflight.(*Group).doCall.func2\n\t/go/pkg/mod/golang.org/x/sync@v0.1.0/singleflight/singleflight.go:189\ngolang.org/x/sync/singleflight.(*Group).doCall\n\t/go/pkg/mod/golang.org/x/sync@v0.1.0/singleflight/singleflight.go:191\ngolang.org/x/sync/singleflight.(*Group).Do\n\t/go/pkg/mod/golang.org/x/sync@v0.1.0/singleflight/singleflight.go:104\ngithub.com/milvus-io/milvus/pkg/util/conc.(*Singleflight[...]).Do\n\t/go/src/github.com/milvus-io/milvus/pkg/util/conc/singleflight.go:18\ngithub.com/milvus-io/milvus/pkg/mq/msgstream/mqwrapper/kafka.(*kafkaClient).getKafkaProducer\n\t/go/src/github.com/milvus-io/milvus/pkg/mq/msgstream/mqwrapper/kafka/kafka_client.go:96\ngithub.com/milvus-io/milvus/pkg/mq/msgstream/mqwrapper/kafka.(*kafkaClient).CreateProducer\n\t/go/src/github.com/milvus-io/milvus/pkg/mq/msgstream/mqwrapper/kafka/kafka_client.go:164\ngithub.com/milvus-io/milvus/pkg/mq/msgstream.(*mqMsgStream).AsProducer.func1\n\t/go/src/github.com/milvus-io/milvus/pkg/mq/msgstream/mq_msgstream.go:108\ngithub.com/milvus-io/milvus/pkg/util/retry.Do\n\t/go/src/github.com/milvus-io/milvus/pkg/util/retry/retry.go:40\ngithub.com/milvus-io/milvus/pkg/mq/msgstream.(*mqMsgStream).AsProducer\n\t/go/src/github.com/milvus-io/milvus/pkg/mq/msgstream/mq_msgstream.go:122\ngithub.com/milvus-io/milvus/internal/rootcoord.newDmlChannels\n\t/go/src/github.com/milvus-io/milvus/internal/rootcoord/dml_channels.go:202\ngithub.com/milvus-io/milvus/internal/rootcoord.newTimeTickSync\n\t/go/src/github.com/milvus-io/milvus/internal/rootcoord/timeticksync.go:121\ngithub.com/milvus-io/milvus/internal/rootcoord.(*Core).initInternal\n\t/go/src/github.com/milvus-io/milvus/internal/rootcoord/root_coord.go:442\ngithub.com/milvus-io/milvus/internal/rootcoord.(*Core).Init.func2\n\t/go/src/github.com/milvus-io/milvus/internal/rootcoord/root_coord.go:512\nsync.(*Once).doSlow\n\t/usr/local/go/src/sync/once.go:74\nsync.(*Once).Do\n\t/usr/local/go/src/sync/once.go:65\ngithub.com/milvus-io/milvus/internal/rootcoord.(*Core).Init\n\t/go/src/github.com/milvus-io/milvus/internal/rootcoord/root_coord.go:511\ngithub.com/milvus-io/milvus/internal/distributed/rootcoord.(*Server).init\n\t/go/src/github.com/milvus-io/milvus/internal/distributed/rootcoord/service.go:217\ngithub.com/milvus-io/milvus/internal/distributed/rootcoord.(*Server).Run\n\t/go/src/github.com/milvus-io/milvus/internal/distributed/rootcoord/service.go:143\ngithub.com/milvus-io/milvus/cmd/components.(*RootCoord).Run\n\t/go/src/github.com/milvus-io/milvus/cmd/components/root_coord.go:51\ngithub.com/milvus-io/milvus/cmd/roles.runComponent[...].func1\n\t/go/src/github.com/milvus-io/milvus/cmd/roles/roles.go:112"]

**Comment by xiaofan-luan** [comment]
@jaime0815 
can we make sure we have sasl support?

**Comment by jaime0815** [comment]
Kafka has already supported SASL, it seems something is broken in 2.3, let me check it.

**Comment by jaime0815** [comment]
@weiZhenkun  it will be fixed on 2.3.1

**Comment by weiZhenkun** [comment]
ok, Thanks, but the pipelines are failed on your PR now? Can you fix it?
